ocaml/testsuite/tests/typing-layouts-float64/alloc.ml

```diff
@@ -0,0 +1,90 @@
+(* TEST
+   flags = "-extension layouts_alpha"
+   * native
+*)
+
+(* A test comparing allocations with unboxed floats to allocations with boxed
+   floats. *)
+
+(* CR layouts v2: test shouldn't print exact allocation numbers. And I've only
+   run it on closure - surely different on other middle ends. *)
+
+
+module type Float_u = sig
+  external to_float : float# -> (float[@local_opt]) = "%box_float"
+  external of_float : (float[@local_opt]) -> float# = "%unbox_float"
+
+  val ( + ) : float# -> float# -> float#
+  val ( - ) : float# -> float# -> float#
+  val ( * ) : float# -> float# -> float#
+  val ( / ) : float# -> float# -> float#
+  val ( ** ) : float# -> float# -> float#
+  val ( > ) : float# -> float# -> bool
+end
+
+module Float_u : Float_u = struct
+  external to_float : float# -> (float[@local_opt]) = "%box_float"
+  external of_float : (float[@local_opt]) -> float# = "%unbox_float"
```

Comment:
This can be just `[@local]`, I think. I don't think anything is gained by using `[@local_opt]`.